code/modules/mob/living/carbon/carbon_update_icons.dm

```diff
@@ -80,7 +80,12 @@
 
 /mob/living/carbon/update_fire(var/fire_icon = "Generic_mob_burning")
 	remove_overlay(FIRE_LAYER)
-	if(on_fire || islava(loc))
+	var/lavaburn = FALSE
+	if(islava(loc))

```

Comment:
Why is this even handled here. Awful

Reply:
t-they did it first! okay okay though, i'll see if i can tie the whole islava part to the lava turf and not update fire